src/views/List.jsx

```diff
@@ -1,6 +1,26 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
+import Button from '../components/Button';
 
 export function List({ data, listToken }) {
+	const [inputItem, setInputItem] = useState('');
+	// This fuction resets the input field:
+	const handleClear = () => {
+		setInputItem('');
+	};
+	const handleInput = (e) => {
+		setInputItem(e.target.value);
+	};
+	// This helper function matches the user input any part of the item name from the items list
+	const stringMatch = (inputItem, listItem) => {
+		return listItem.toLowerCase().includes(inputItem.toLowerCase());
+	};
+
+	// Applies the helper function to narrow down the data
+	const filterData = data.filter((listItem) =>
```

Comment:
Just a thought here, because I'm not certain my suggestion is correct: since multiple users can update the same list, would `useEffect` make sense here? To make sure that `filteredData` is always in-sync with the data?

Reply:
Corina, thank you for the feedback! I am not sure if a useEffect hook would sync the list items in this case because the list data is being passed on to List.jsx as a prop from App.jsx. Although I am not certain either, we can definitely discuss this during the office hour tomorrow.

Reply:
The data is returned by the `useShoppingListData` function in `firebase.js`, and then via the App makes its way as a prop to the components that use it. I thought since there could be multiple users modifying the same list, we would need `useEffect` to mitigate the different updates from other users. But I had forgotten that `useShoppingListData` already has a `useEffect` to update the state in real-time, so a second one would definitely be an overkill (I think some people actually talk about the a `useEffect` epidemic 😂 )

Reply:
Great discussion here, @CorinaMurg & @AslanTaheri ! 👏